src/scheduler/Executor.cpp

```diff
@@ -66,14 +66,18 @@ void Executor::finish()
         threadTaskQueues[i].enqueue(
           ExecutorTask(POOL_SHUTDOWN, nullptr, nullptr, false, false));
 
+        faabric::util::UniqueLock threadsLock(threadsMutex);
+        // copy shared_ptr to avoid racing
```

Comment:
Style in the repo is to start comments with capital letters (but unfortunately couldn't find a rule to enforce it).